### ISSUE TITLE
Running from Source Feedback from a layman (in this case also probably an idiot)

### ISSUE THREAD

**Issue by Digo08** [ArchipelagoMW/Archipelago#1196]
### What task needs to be completed?

TLDR- for the layman with little to no experience this document is useless.  I didn't know what or why I would need to go through these steps.  My guess is that this will be more for people who do have something of a programing and or Linux background, because it took my middle ground friend who knows just enough to explain why I had no idea what you were asking for on discord.  Below i used [ ] to show the area i'm talking about then **  ** for my input.

Python 3.8.7 or newer                                                        **(regular person would probable know this is a programming language)**
pip (Depending on platform may come included)             **(googled this and figured out I did know it to an extent, but didn't 
                                                                                      know what "pip" meant before I googled what is a pip in programming) **

A C compiler
possibly optional, read OS-specific sections                 **(I know what a compiler is, but don't know what an A C compiler is.  Also 
                                                                                   may be a stretch, but not everyone refers to an operating system as an OS, 
                                                                                        so may as well type it out)**

[Then run any of the starting point scripts, like Generate.py, and the included ModuleUpdater should prompt to install or update the required modules and after pressing enter proceed to install everything automatically. After this, you should be able to run the programs.

With yaml(s) in the Players folder, Generate.py will generate the multiworld archive.  **understood primarily because i've been playing archipelago for quite some time now**
MultiServer.py, with the filename of the generated archive as a command line parameter, will host the multiworld locally.
--log_network is a command line parameter useful for debugging.  **Again, means nothing to me.**

WebHost.py will host the website on your computer.
You can copy docs/webhost configuration sample.yaml to config.yaml to change WebHost options (like the web hosting port number).
As a side effect, WebHost.py creates the template yamls for all the games in WebHostLib/static/generated.]  **This last paragraph again means nothing to me.**

[Windows
Recommended steps

Download and install a "Windows installer (64-bit)" from the [Python download page](https://www.python.org/downloads)

Download and install full Visual Studio from [Visual Studio Downloads](https://visualstudio.microsoft.com/downloads/) or an older "Build Tools for Visual Studio" from [Visual Studio Older Downloads](https://visualstudio.microsoft.com/vs/older-downloads/).

Refer to [Windows Compilers on the python wiki](https://wiki.python.org/moin/WindowsCompilers) for details
This step is optional. Pre-compiled modules are pinned on [Discord in #archipelago-dev](https://discord.com/channels/731205301247803413/731214280439103580/905154456377757808)
It is recommended to use [PyCharm IDE](https://www.jetbrains.com/pycharm/)

Run Generate.py which will prompt installation of missing modules, press enter to confirm]  **This is simple enough as you're telling me i need these programs and most people would be comfortable going ahead and just downloading them as that's what they're being told to do for the game**

[macOS
Refer to [Guide to Run Archipelago from Source Code on macOS](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/generic/docs/mac_en.md).]  **This is fine i'm sure for those that do mac.  I do not so i can't really give input.**

Optional: A Link to the Past Enemizer
Only required to generate seeds that include A Link to the Past with certain options enabled. You will receive an error if it is required.   **Include Zelda in from of A Link to the Past as those that don't play it may not know what that is game wise.**

[You can get the latest Enemizer release at [Enemizer Github releases](https://github.com/Ijwu/Enemizer/releases). It should be dropped as "EnemizerCLI" into the root folder of the project. Alternatively, you can point the Enemizer setting in host.yaml at your Enemizer executable.]   **Honestly doesn't mean much as i don't know what an enemizer is.  If my group were to play it i'd have to go to the discord and get help.**

[Optional: SNI
SNI is required to use SNIClient. If not integrated into the project, it has to be started manually.

You can get the latest SNI release at [SNI Github releases](https://github.com/alttpo/sni/releases). It should be dropped as "SNI" into the root folder of the project. Alternatively, you can point the sni setting in host.yaml at your SNI folder.]   **Again, had to google SNI as i don't know what it is.  At least add in the name the first time before using the acronym.**

[Running tests
Run pip install pytest pytest-subtests, then use your IDE to run tests or run pytest from the source folder.]  **I don't know what an IDE is.**

**Comment by black-sliver** [comment]
Since there was no more feedback, and I think it should mostly be fine for most purposes, this can probably be closed.
